### ISSUE TITLE
Fela is overriding styles as components get rendered

### ISSUE THREAD

**Issue by SpencerCDixon** [rofrischmann/fela#428]
**Version**: 6.0.2
**Environment**: DOM
**Type**: Bug / General

-------------------
#### Description
Something really weird is happening with Fela in react storybook.  As I click on various stories and render new components the other components styles get overridden. 

See attached gif:

![fela-override](https://user-images.githubusercontent.com/7471018/31515485-53ab4bf0-af63-11e7-8417-6911fe3941fc.gif)

Button component:
```javascript
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { combineRules, createComponent } from '@covered/rio-style-fela';

const raised = ({ raised = false, theme: { shadow } }) => ({
  boxShadow: raised ? shadow.button : 'none',
});

const color = ({ secondary, emphasis, theme: { colors } }) => ({
  background: secondary
    ? colors.secondary
    : emphasis ? colors.emphasis : colors.primary,
});

const base = ({ theme }) => ({
  color: theme.colors.white,
  padding: '10px 14px',
  border: 'none',
  borderRadius: 100,
  letterSpacing: 1,
  fontFamily: theme.fonts.primary,
  fontWeight: theme.fonts.bold,
});

const rules = combineRules(base, raised, color);

export default createComponent(rules, 'button');
```

Provider:
```javascript
import React, {
  Component,
  Children,
  isValidElement,
  cloneElement,
} from 'react';
import PropTypes from 'prop-types';
import createRenderer from './createRenderer';
// import { render } from 'fela-dom';
import { Provider } from 'react-fela';
import { theme } from '@covered/rio-style-const';
import { ThemeProvider } from '@covered/rio-style-fela';

class StyleProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    // Inject override
    renderer: PropTypes.object,
    // Renderer environment settings
    dev: PropTypes.bool,
    perf: PropTypes.bool,
    debug: PropTypes.bool,
  };

  render() {
    const { renderer, children, dev, perf, debug, ...restProps } = this.props;
    const providerRenderer = renderer || createRenderer({ dev, perf, debug });
    const child = Children.only(children);

    return (
      <Provider renderer={providerRenderer}>
        <ThemeProvider theme={theme}>
          {isValidElement(child)
            ? cloneElement(child, { ...restProps })
            : child}
        </ThemeProvider>
      </Provider>
    );
  }
}

export default StyleProvider;
```

**Comment by rofrischmann** [comment]
Can you please give me a working repository? I got no time to create an example myself with just those snippets. I feel like it's the same issue as #412. If I got a example repo, I can test it and maybe fix for both.

**Comment by SpencerCDixon** [comment]
Yeah don't worry about it, it's definitely something I'm doing wrong will close these for now.

**Comment by rofrischmann** [comment]
I dont think so. This might be an actual issue! If you could ping me on Gitter, i might be able to help

**Comment by SpencerCDixon** [comment]
Hey @rofrischmann the problem was in this line:

```
    const providerRenderer = renderer || createRenderer({ dev, perf, debug });
```

If you look above my `StyleProvider` was creating a new renderer on every render so when I looked at new 'stories' in react story it was setting all the styles of the other stories to the latest one I viewed.

Once I created a single renderer and passed it in as a prop everything was fixed.  I think that is what was going on but either way it works now!
